core/src/main/java/org/elasticsearch/bootstrap/Check.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.bootstrap;
+
+/**
+ * Encapsulates a bootstrap check.
+ */
+public interface Check {
```

Comment:
I wonder if this class should be renamed to `BootstrapCheck` and the currently-existing `BootstrapCheck` should be renamed to `BootstrapChecks` (or something like that, basically looking to give `Check` a more indicative name now that it is not an nested class in `BootstrapCheck`)?